File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i not in "+-*/":
                stack.append(int(i))
                continue
            
         
            
            num2 = stack.pop()
            num1 = stack.pop()
            
            if i == "+":
                newVal = num1 + num2
            if i == "-":
                newVal = num1 - num2
            if i == "*":
                newVal = num1 * num2
            if i == "/":
                newVal = int(num1 / num2)
                    
            stack.append(newVal)
                    
        return stack.pop()
                        
        

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    /*
    tokens = ["4","13","5","/","+"]
    
    4 + (13/5)
    
    1. init a stack
    2. traverse the input arr to put elements
        2.1) if element is found, then put into stack
        2.2) if operator is found, then pop the two recent values from stack and save as num2, num1
                num1 operator num2
                and put back into the new value into stack
    
    */
    let stack = [];
    const OPERATIONS = ['+', '-','*', '/'];
    for(let token of tokens){
        if(!OPERATIONS.includes(token)){
            stack.push(Number(token));
            continue;
        }
        
        const n2 = stack.pop();
        const n1 = stack.pop();
        
        switch(token){
            case '+':
                stack.push(n1+n2);
                break;
            case '-':
                stack.push(n1-n2);
                break;
            case '*':
                stack.push(n1*n2);
                break;
            case '/':
                stack.push(Math.trunc(n1/n2));
                break;
            default:
                break;
        }
    }
    return stack.pop()
};